zyte_spider_templates/_incremental/manager.py

```diff
@@ -22,11 +23,19 @@
 logger = logging.getLogger(__name__)
 
 INCREMENTAL_SUFFIX = "_incremental"
+_MAX_LENGTH = 2048 - len(INCREMENTAL_SUFFIX)
 COLLECTION_API_URL = "https://storage.scrapinghub.com/collections"
 
 THREAD_POOL_EXECUTOR = ThreadPoolExecutor(max_workers=10)
 
 
+def _get_collection_name(crawler: Crawler) -> str:
+    if name := crawler.settings.get("INCREMENTAL_CRAWL_COLLECTION_NAME"):
+        return name
+    name = get_spider_name(crawler).rstrip("_")[:_MAX_LENGTH] + INCREMENTAL_SUFFIX
+    return slugify(name, separator="_", lowercase=False, regex_pattern=r"[^a-zA-Z0-9_]")
```

Comment:
We could replace any non-valid character with `_`. It would be trivial and reliable. I like the å → a conversion of slugify and lower-level libraries like unidecode, but here it makes sense to prioritize future-proofing.